src/pybind/mgr/dashboard/module.py

```diff
@@ -255,6 +255,10 @@ def fs_status(self, fs_id):
                 filesystem = fs
                 break
 
+        if filesystem is None:
+            raise cherrypy.HTTPError(404,
```

Comment:
I'd raise http exceptions as late as possible, i.e. this one in the filesystem function.